beamer/state_machine.py

```diff
@@ -272,11 +272,14 @@ def _handle_claim_made(event: ClaimMade, context: Context) -> HandlerResult:
     claim = context.claims.get(event.claim_id)
 
     if claim is None:
+        if event.last_challenger != ADDRESS_ZERO:
+            return False, None
+
```

Comment:
The commit message of `f19f7c434081f1402c77fa95762c01600a758740` says " agent: Only withdraw if there are no ClaimMade events in loop " but I do not understand how this relates to the above change. Could you please explain?

Reply:
The relation is the following: 

There is a change above which don't processes ClaimMade Events of the agent if the fill isnt there yet. If that is the case then a second (or third, etc) ClaimMade event will be processed before the first one. 
We need to also push back those events, namely when a claim is no created yet, but we receive ClaimMade events with challengers in it. This means the initital ClaimMade event must have been pushed back, so also push back the others

Reply:
Ah, that totally explains it. Thank you for such a nice explanation!